sdks/java/io/thrift/src/test/java/org/apache/beam/sdk/io/thrift/ThriftIOTest.java

```diff
@@ -0,0 +1,233 @@
+import org.apache.beam.repackaged.core.org.apache.commons.lang3.RandomUtils;
+import org.apache.beam.sdk.coders.StringUtf8Coder;
+import org.apache.beam.sdk.io.FileIO;
+import org.apache.beam.sdk.testing.PAssert;
+import org.apache.beam.sdk.testing.TestPipeline;
+import org.apache.beam.sdk.transforms.Create;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.vendor.guava.v26_0_jre.com.google.common.collect.ImmutableList;
+import org.apache.beam.vendor.guava.v26_0_jre.com.google.common.io.Resources;
+import org.apache.thrift.protocol.TBinaryProtocol;
+import org.apache.thrift.protocol.TCompactProtocol;
+import org.apache.thrift.protocol.TJSONProtocol;
+import org.apache.thrift.protocol.TProtocolFactory;
+import org.apache.thrift.protocol.TSimpleJSONProtocol;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Tests for {@link ThriftIO}. */
+@RunWith(JUnit4.class)
+public class ThriftIOTest implements Serializable {
+
+  private static final String RESOURCE_DIR = "ThriftIOTest/";
+
+  private static final String THRIFT_DIR = Resources.getResource(RESOURCE_DIR).getPath();
+  private static final String ALL_THRIFT_STRING =
+      Resources.getResource(RESOURCE_DIR).getPath() + "*";
+  private static final TestThriftStruct TEST_THRIFT_STRUCT = new TestThriftStruct();
+  private static List<TestThriftStruct> testThriftStructs;
+  private final TProtocolFactory tBinaryProtoFactory = new TBinaryProtocol.Factory();
+  private final TProtocolFactory tJsonProtocolFactory = new TJSONProtocol.Factory();
+  private final TProtocolFactory tSimpleJsonProtocolFactory = new TSimpleJSONProtocol.Factory();
+  private final TProtocolFactory tCompactProtocolFactory = new TCompactProtocol.Factory();
+  @Rule public transient TestPipeline mainPipeline = TestPipeline.create();
+  @Rule public transient TestPipeline readPipeline = TestPipeline.create();
+  @Rule public transient TestPipeline writePipeline = TestPipeline.create();
+  @Rule public transient TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+  @Before
+  public void setUp() throws Exception {
+    byte[] bytes = new byte[10];
+    ByteBuffer buffer = ByteBuffer.wrap(bytes);
+
+    TEST_THRIFT_STRUCT.testByte = 100;
+    TEST_THRIFT_STRUCT.testShort = 200;
+    TEST_THRIFT_STRUCT.testInt = 2500;
+    TEST_THRIFT_STRUCT.testLong = 79303L;
+    TEST_THRIFT_STRUCT.testDouble = 25.007;
+    TEST_THRIFT_STRUCT.testBool = true;
+    TEST_THRIFT_STRUCT.stringIntMap = new HashMap<>();
+    TEST_THRIFT_STRUCT.stringIntMap.put("first", (short) 1);
+    TEST_THRIFT_STRUCT.stringIntMap.put("second", (short) 2);
+    TEST_THRIFT_STRUCT.testBinary = buffer;
+
+    testThriftStructs = ImmutableList.copyOf(generateTestObjects(1000L));
+  }
+
+  /** Tests {@link ThriftIO#readFiles(Class)} with {@link TBinaryProtocol}. */
+  @Test
```

Comment:
There doesn't seem to be a test for `read().from(...)` - should there be one? (and it's okay if the answer is 'don't need one') : )

Reply:
`read` was in the old implementation and I will remove the references to it. I think that `readFiles()` will cover most use cases for this IO.